partiql-lang/src/main/kotlin/org/partiql/lang/planner/GlobalVariableResolver.kt

```diff
@@ -17,6 +18,21 @@ sealed class GlobalResolutionResult {
      */
     data class GlobalVariable(val uniqueId: String) : GlobalResolutionResult()
```

Comment:
A change I've been wanting to make forever is to make `uniqueId` here an `IonElement` so we can put arbitrary Ion data here instead of having to "abuse" this field by putting Ion-text here instead.  I'll leave a note in the main PIG domain where we need to change this as well.... EDIT: github won't let me comment there--to do this we would also need to change `partiql_logical_resolved.expr.global_id.unique_id` to the `ion` type.

Since we're the API anyway, now might be a good time to do that as well...